### PR TITLE
add update to info

### PR DESCRIPTION
**Pull request by joamatab**

### CODE REVIEW

**Comment by sebastian-goeldi** [comment]
What's the reason for needing this?

**Comment by joamatab** [comment]
we do a lot of this in gdsfactory 

```python
c.info.update(**info_settings)
```

**Comment by sebastian-goeldi** [comment]
Then you only need the dict and not the info object one, right?

That makes this way easier. If you don't need iterable and Info object, I don't see why we should add it here. It just makes the signature unreadable.

**Comment by joamatab** [comment]
Yes, good point

File: src/kfactory/kcell.py [hunk 1]
```diff
@@ -220,6 +220,19 @@ def __setitem__(
...
+                self.__setitem__(key, value)
+        elif isinstance(data, Info):
+            for key, value in data.model_dump().items():
+                self.__setitem__(key, value)
+        elif isinstance(data, Iterable):
+            for key, value in data:
+                self.__setitem__(key, value)
+        else:
+            raise TypeError("Unsupported data type for update")
+
```
Comment (sebastian-goeldi) [inline src/kfactory/kcell.py:235]:
```suggestion
    def update(self, data: dict[str, bool | int | float | str]) -> None:
        for key, value in data.items():
            settatr(self, key, value)
        
    def __iadd__(self, other: Info):
        for key, value in other.items():
            setattr(self, key, value)
        for key, value in other.model_dump().items():
            setattr(self, key, value)
```

**Comment by joamatab** [comment]
Looks good to me,

feel free to merge

**Comment by joamatab** [comment]
included in this one

https://github.com/gdsfactory/kfactory/pull/273

**Pull request closed without merging**